### PR TITLE
Docs: adding data sorting tutorial (#746)

### PR DESCRIPTION
**Pull request by kirszenbaum**
From what I gather, there's no native sorting as such, but now, instead of `moveRows` you can use `setRowOrder` and `setColumnOrder`.

@izulin @wojciechczerniak - can I ask for your help with updating the data sorting demo?

- [x] Updated a few comments in the `setRowOrder` and `setColumnOrder` API ref
- [x] Added "Reordering rows" and "Reordering columns" sections to the "Basic operations" page
- [x] Rewrote the "Sorting data" page

Any repetitions are purposeful (each procedure section should make sense on its own).

### CODE REVIEW

File: docs/guide/basic-operations.md [hunk 1]
```diff
@@ -201,6 +215,20 @@ this operation together with their absolute addresses and new values.
...
+You can change the order of columns by using the `setColumnOrder` method. You need to pass the following parameters:
+* Sheet ID
+* New column order
+
+This method returns a list of cells whose values were affected by
+this operation, together with their absolute addresses and new values.
+
+```javascript
+// column 0 and column 2 swap places
+const changes = hfInstance.setRowOrder(0, [2, 1, 0]);
```
Comment (izulin) [inline docs/guide/basic-operations.md:229]:
```suggestion
const changes = hfInstance.setColumnOrder(0, [2, 1, 0]);
```

File: docs/guide/sorting-data.md [hunk 2]
```diff
@@ -1,17 +1,202 @@
 # Sorting data
 
+In HyperFormula, you can sort data by reordering rows and columns.
+
+## Sorting data in HyperFormula
+
+Sorting data in HyperFormula works similarly to Google Sheets and Microsoft Excel:
+
+1. First, HyperFormula sorts formulas by their results (calculated values).
+2. When the sorting is done, each formula is recalculated with regard to its new position.
```
Comment (wojciechczerniak) [inline docs/guide/sorting-data.md:10]: Nope, that's description for Handsontable sorting. HF does not compare data, you just provide new order. How it was calculated is up to the user

File: docs/guide/sorting-data.md [hunk 3]
```diff
@@ -1,17 +1,202 @@
...
+//   newValue: 5,
+// }]
+```
+
+## Sorting columns
+
+To sort columns, use the [`isItPossibleToSetColumnOrder`](../api/classes/hyperformula.md#isitpossibletosetcolumnorder) and [`setColumnOrder`](../api/classes/hyperformula.md#setcolumnorder) methods.
+
+### Step 1: Choose a new column order
+Choose your required permutation of column numbers.
```
Comment (wojciechczerniak) [inline docs/guide/sorting-data.md:110]: Maybe indexes or positions would be more precise?

**Review by wojciechczerniak — CHANGES_REQUESTED** [review]

File: docs/guide/sorting-data.md (hunk 3 above)
Comment (kirszenbaum) [reply docs/guide/sorting-data.md]: @wojciechczerniak - yup, I was thinking of indexes, but wasn't sure. Done: 9d9be0e

File: docs/guide/sorting-data.md (hunk 2 above)
Comment (kirszenbaum) [reply docs/guide/sorting-data.md]: @wojciechczerniak - ok, makese sense - so the previous intro was just a placeholder then. How about this: 9df4703

Comment (wojciechczerniak) [reply docs/guide/sorting-data.md]: Perfect 👍

**Review by wojciechczerniak — APPROVED** [review]
🏆

**Review by izulin — APPROVED** [review]

**Pull request merged**